### DISCUSSION TITLE
Fronius Primo Register documentation

### DISCUSSION THREAD

**Discussion by petebitt99** [amberelectric/public-api#197] [Q&A]
I am hoping to automate the control of my Fronius Primo Inverter, a lot like many of you using HA. I will use the Amber API in conjunction with my local Tesla API as the basis of the data I need to automate the management of negative FIT's. 
I am after the documentation that includes the registers needed to be read/written and reset to do this. I will be communicating with the inverter using Modbus in a similar way to Village Energy.
Could anyone point me in the right direction?  I'm sure from what I read here that others are doing this so must know these details.

**Comment by icenov** [comment]
I did use modbus with my Primo about 6 years ago, and will try and find the files. In the meantime read this as it was very helpful to me at the time.
https://stoborblog.wordpress.com/2017/03/11/sunspec-solar-viewer-fronius-modbus/

**Comment by petebitt99** [comment]
Thanks very much Tony.PeterOn 20 Mar 2024, at 9:58 AM, Tony Vassallo ***@***.***> wrote:﻿
I did use modbus with my Primo about 6 years ago, and will try and find the files. In the meantime read this as it was very helpful to me at the time.
https://stoborblog.wordpress.com/2017/03/11/sunspec-solar-viewer-fronius-modbus/

**Reply by icenov** [reply]
I couldn't find any Primo files, but do have a spreadsheet for the Symohybrid with a lot of detail. From memory, I think the registers are the same for Primo. 
[Inverter_Register_Map_Int&SF_v1.0__with_SYMOHYBRID_MODEL_124.xlsx](https://github.com/amberelectric/public-api/files/14766357/Inverter_Register_Map_Int.SF_v1.0__with_SYMOHYBRID_MODEL_124.xlsx)

**Comment by petebitt99** [comment]
Tony,

You are right that’s the one.  Thanks so much for going to the trouble. To tell you the truth I had forgotten I’d asked you I have been chasing my tail so much on this one.  In the end the issue was that my inverter was still set to float. I had gone in and changed to INT+SF but I couldn’t have saved it properly.  When I ended up reading register 40070 and got 111 I knew what the problem was. It is now working.  

Peter
